shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -355,6 +355,61 @@ static FlutterAutofillType autofillTypeOf(NSDictionary* configuration) {
   return FlutterAutofillTypeNone;
 }
 
+static BOOL isApproximatelyEqual(float x, float y, float delta) {
+  return x >= y - delta && x <= y + delta;
+}
+
+// Checks whether point should be considered closer to selectionRect compared to
+// otherSelectionRect.
+//
+// If checkRightBoundary is set, the right-center point on selectionRect and
+// otherSelectionRect will be used instead of the left-center point.
+//
+// This uses special (empirically determined) logic for determining the closer rect,
+// rather than a simple distance calculation. First, the closer vertical distance is
+// determined. Within the closest y distance, if the point is above the bottom
+// of the closest rect, the x distance will be minimized; however, if the point is
+// below the bottom of the rect, the x value will be maximized.
```

Comment:
It sounds like you've tried this natively on an iPad and this is what you've observed the behavior to be, rather than just a pure distance calculation?  If that's true, consider adding the exact device and iOS version that you used to determine this behavior to this comment.  Maybe `empirically determined on an ... running ...`.  That way if someone in the future notices that it doesn't perfectly match native, they have some idea of where this algorithm came from.